symengine/pow.cpp

```diff
@@ -315,28 +315,28 @@ RCP<const Basic> exp(const RCP<const Basic> &x)
 Log::Log(const RCP<const Basic> &arg)
     : arg_{arg}
 {
-    SYMENGINE_ASSERT(is_canonical(arg))
+    SYMENGINE_ASSERT(arg != null)
```

Comment:
The same here. I know that you just ported the null checks from the `is_canonical`, but it was there for historical reasons, I think even there it is not needed.